scrapers/distgit.py

```diff
@@ -0,0 +1,199 @@
+# SPDX-License-Identifier: GPL-3.0+
+
+from __future__ import unicode_literals
+import re
+
+from builtins import bytes
+from bs4 import BeautifulSoup
+from requests import ConnectionError
+
+from scrapers.base import BaseScraper
+from scrapers.utils import retry_session
+from purview.utils.general import timestamp_to_datetime
+from purview.models.distgit import DistGitCommit, DistGitPush, DistGitBranch, DistGitRepo
+from purview.models.bugzilla import BugzillaBug
+from purview.models.user import User
+from purview import log
+
+
+class DistGitScraper(BaseScraper):
+    cgit_url = 'http://pkgs.devel.redhat.com/cgit/'
+    # The tuple of namespaces to try when determining which namespace this git module belongs
+    # to since this information isn't stored in GitBZ yet
+    namespaces = ('rpms', 'containers', 'modules', 'tests')
+
+    def run(self, since=None):
+        """
+        Main function that runs the dist-git scraper
+        :param since: a string representing a datetime to start scraping data from
+        :return: None
+        """
+        if since is None:
+            dt_since = self.default_since
+        else:
+            dt_since = timestamp_to_datetime(since)
+        results = self.get_distgit_data(dt_since)
+        counter = 0
+        for result in results:
+            counter += 1
+            log.info('Processing commit and push entry {0}/{1}'.format(
+                str(counter), str(len(results))))
+            repo_info = self.get_repo_info(result['module'], result['sha'])
+            if not repo_info:
+                log.info('Skipping nodes creation with commit ID {0} and push ID {1}'.format(
+                    result['commit_id'], result['push_id']))
+                continue
+
+            log.debug('Creating nodes associated with commit ID {0} and push ID {1}'.format(
+                result['commit_id'], result['push_id']))
+            repo = DistGitRepo.get_or_create({
+                'namespace': repo_info['namespace'],
+                'name': result['module']
+            })[0]
+            branch_name = result['ref'].rsplit('/', 1)[1]
+            branch = DistGitBranch.get_or_create(
```

Comment:
Maybe we should keep all the stuff related to uploading to Neo4j in one method just like in koji and bugzilla scrapers? Let's keep run() small